src/db/couchDevices.ts

```diff
@@ -218,3 +281,47 @@ function makeDeviceRow(
     }
   }
 }
+
+export const streamDevicesByLocation = async (
```

Comment:
This isn't a stream - it will return everything in the database in one big blob, which will crash without further attention. Instead of relying on the caller to do `viewToStream`, we should implement that internally, making this function safe to use anywhere:

```ts
export async function*  streamDevicesByLocation(
  connection: ServerScope,
   location: { country?: string; region?: string; city?: string }
): AsyncIterableIterator<AccessLog> {
 ...
  for await (const row of viewToStream(...)) {
    yield unpackDevice(row)
  }
}
```